pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperKitCreationService.java

```diff
@@ -0,0 +1,276 @@
+            return new KitResponse(ADDRESS_VALIDATION_ERROR, juniperKitRequest.getJuniperKitId(), null);
+        }
+
+        ArrayList<KitRequest> orderKits = new ArrayList<>();
+
+
+        SimpleResult result = TransactionWrapper.inTransaction(conn -> {
+            SimpleResult transactionResults = new SimpleResult();
+            return createKit(ddpInstance, kitType, juniperKitRequest, kitRequestSettings, easyPostUtil, kitTypeName, orderKits, conn,
+                    transactionResults);
+
+            //            only order if external shipper name is set for that kit request, not needed for now
+            //            if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+            //                return orderExternalKits(kitRequestSettings, orderKits, easyPostUtil, shippingCarrier, conn);
+            //            }
+        });
+
+        if (result.resultException != null) {
+            log.error(String.format("Unable to create Juniper kit for %s", juniperKitRequest), result.resultException);
+            return new KitResponse(DSM_ERROR, juniperKitRequest.getJuniperKitId(), result.resultException);
+
+        }
+
+        log.info(juniperKitRequest.getJuniperKitId() + " " + ddpInstance.getName() + " " + kitTypeName + " kit created");
+        return new JuniperKitStatus(null, null, null);
+    }
+
+    /**
+     * checkAddress tries creating an address in EasyPost. If it is successful,
+     * sets the kit's easypostAddressId and returns true, if not returns false
+     * An address is valid only if participant has shortId, first - and lastName, for Juniper shortId is the juniperParticipantId
+     *
+     * @param juniperKitRequest the JuniperKitRequest with address to check
+     */
+
+    public boolean checkAddress(JuniperKitRequest juniperKitRequest, String phone, EasyPostUtil easyPostUtil) {
+        if ((StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID()))
+                || StringUtils.isBlank(juniperKitRequest.getLastName())) {
+            return false;
+        }
+        //let's validate the participant's address
+        String name = "";
+        if (StringUtils.isNotBlank(juniperKitRequest.getFirstName())) {
+            name += juniperKitRequest.getFirstName() + " ";
+        }
+        name += juniperKitRequest.getLastName();
+        if (juniperKitRequest.isSkipAddressValidation()) {
+            try {
+                Address address = easyPostUtil.createBroadAddress(name, juniperKitRequest.getStreet1(), juniperKitRequest.getStreet2(),
+                        juniperKitRequest.getCity(),
+                        juniperKitRequest.getPostalCode(), juniperKitRequest.getState(), juniperKitRequest.getCountry(), phone);
+                juniperKitRequest.setEasypostAddressId(address.getId());
+                return true;
+            } catch (EasyPostException e) {
+                log.warn("Easypost couldn't create an address for " + juniperKitRequest.getShortId());
```

Comment:
Unclear what's going on here. Can you add a comment?
Will we get this exception for a bad request, or is this more like an internal error?
Why are we logging at warning level?
The results of the stack trace are going, I assume, to stdout or stderr, which may or may not be logged.

Reply:
addressed